File: unity/Assets/Scripts/Multiplayer/Rooms/RoomListRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms
{
    /// <summary>
    /// Request data for listing game rooms with optional filtering in Unity.
    /// </summary>
    public class RoomListRequest
    {
        /// <summary>
        /// Search term to filter rooms by name.
        /// </summary>
        public string search = "";

        /// <summary>
        /// Maximum number of rooms to return.
        /// </summary>
        public int limit = 20;
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/RoomListRequest.java
package com.michitai.multiplayer.rooms;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Request for listing available game rooms.
 */
public class RoomListRequest {
    @JsonProperty("search")
    private String search;

    @JsonProperty("limit")
    private Integer limit;

    public RoomListRequest(String search, Integer limit) {
        this.search = search;
        this.limit = limit;
    }

    public String getSearch() {
        return search;
    }

    public void setSearch(String search) {
        this.search = search;
    }

    public Integer getLimit() {
        return limit;
    }

    public void setLimit(Integer limit) {
        this.limit = limit;
    }
}
